### PR TITLE
common: bwl: dwpal: Fix DWPAL compilation issues introduced from refa…

### PR DESCRIPTION
**Pull request by amdickow**
…ctoring

This commit fixes the compilation of BWL for Linux host/target platform using
BWL_TYPE=DWPAL instead of DUMMY.

The following refactoring commits, introduced a couple of regressions in
and around the dwpal implementation:

common: bwl: use beerocks::net::sScanResult
da6b2c0feab9dd7d929d2493d2b4e6fe961489eb

common: refactor net_struct sMacAddr
397254babb09ddb4bd362c88317694198f7d65ac

### CODE REVIEW

**Comment by tomereli** [comment]
Travis failed due to clang-format issues:
image
Please run `./tools/docker/clang-format.sh` on each commit, otherwise CI will always block the PR.

File: common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp
Comment (tomereli) [inline common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp:85]: We would normally split this into 2 commits if you decided to fix whitespace issues - one for the actual change, and one for the whitespace fixes. However this is not crucial, just a comment - act on it or don't act on it, your choice (nitpicking)

(We mark comments which are not blockers for merge like this, @arnout uses a different OK icon)

**Review by tomereli — APPROVED** [review]
LGTM (Other than the whitespace changes which belong in a separate commit but that's not a merge blocker)

Just fix the clang-format issue and you can merge the PR.

File: common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp
Comment (amdickow) [reply common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp:85]: Got it, thanks for the comment.
Let me just push a commit on top, to have the CI job run go green before merging.

**Comment by tomereli** [comment]
CI passed, you can go ahead and squash the last commit and merge

**Pull request merged**
